node/bft/src/primary.rs

```diff
@@ -822,10 +822,6 @@ impl<N: Network> Primary<N> {
                     }
                     // Retrieve the committee lookback for the round.
                     let committee_lookback = self_.ledger.get_committee_lookback_for_round(proposal.round())?;
-                    // Retrieve the address of the validator.
-                    let Some(signer) = self_.gateway.resolver().get_address(peer_ip) else {
```

Comment:
I don't believe this code is redundant. The previous check above:
```
        if self.gateway.resolver().get_address(peer_ip).map_or(true, |address| address != signer) {
            // Proceed to disconnect the validator.
            self.gateway.disconnect(peer_ip);
            bail!("Malicious peer - batch signature is from a different validator ({signer})");
        }
```

Returns true if there was no signer associated with a connected peer ip. Removing this check means that we would now allow validators to process the signatures for non-connected validators.

In theory, could alter the above check, but I believe the double-lookup is cheap enough to not warrant any changes.